templates/lib/validatorset.js

```diff
@@ -1,8 +1,11 @@
 var Joi = require('joi');
 
 var fieldValidators = {
-  username:       Joi.string().alphanum().min(1).required().label('Username'),
-  password:       Joi.string().min(8).required().label('Password'),
+  username:       Joi.alternatives().try(
+                    Joi.string().alphanum().min(1).required(),
+                    Joi.string().email().required())
+                  .label('Username or email'),
+  password:       Joi.string().regex(/^\S{8,16}$/).label('Password'),
```

Comment:
Because I don't really know anything about this `joi` library, I don't feel particularly qualified to R+ this part... And since I don't actually know the context of the rest of the codebase I'm not sure if I should R+ either? Maybe @jbuck or @cadecairos or someone else w/ a better understanding of it can review?